atom/browser/ui/message_box_mac.mm

```diff
@@ -192,21 +145,17 @@ void ShowMessageBox(NativeWindow* parent_window,
     int ret = [[alert autorelease] runModal];
     callback.Run(ret, alert.suppressionButton.state == NSOnState);
   } else {
-    ModalDelegate* delegate = [[ModalDelegate alloc] initWithCallback:callback
-                                                             andAlert:alert
-                                                         callEndModal:false];
-
     NSWindow* window =
         parent_window ? parent_window->GetNativeWindow().GetNativeNSWindow()
                       : nil;
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wdeprecated-declarations"
-    [alert
-        beginSheetModalForWindow:window
-                   modalDelegate:delegate
-                  didEndSelector:@selector(alertDidEnd:returnCode:contextInfo:)
-                     contextInfo:nil];
-#pragma clang diagnostic pop
+    // Duplicate the callback object here since c is a reference and gcd would
+    // only store the pointer, by duplication we can force gcd to store a copy.
```

Comment:
I don't suppose this is something that we could do with the standard c++ syntax for block binding? e.g.

```mm
[alert beginSheetModalForWindow:window 
              completionHandler:^[&, callback](NSModalResponse response) {
                // ...
              }]
```

Reply:
@nornagon I copied this from another `completionHandler` usage in our code.  Seem to remember this being the best workaround back when we wrote that.  I tried binding like your comment and it crashes 😢

Reply:
The binding is possible, the variable should be captured by value rather than by reference

```
completionHandler: [=](NSModalResponse response) {}
```

Reply:
@deepak1556 If this is the case I'll update these _everywhere_ in our code in a follow up PR